### ISSUE TITLE
[Event Request] table 77 "Report Selections" - OnEnqueueMailingJobOnBeforeEnqueueJobQueueEntry

### ISSUE THREAD

**Issue by pri-kise** [microsoft/ALAppExtensions#16962]
Can we please get the following event?
```al
local procedure EnqueueMailingJob(RecordIdToProcess: RecordID; ParameterString: Text; Description: Text)
var
    JobQueueEntry: Record "Job Queue Entry";
    EmailFeature: Codeunit "Email Feature";
    MaxNumberOfRetries: Integer;
begin
    MaxNumberOfRetries := 3;
    if EmailFeature.IsEnabled() then // Avoid multiple failed entries in the Email Outbox if the JQ fails
        MaxNumberOfRetries := 0; // So that the job runs only once

    JobQueueEntry.Init();
    JobQueueEntry."Object Type to Run" := JobQueueEntry."Object Type to Run"::Codeunit;
    JobQueueEntry."Object ID to Run" := CODEUNIT::"Document-Mailing";
    JobQueueEntry."Job Queue Category Code" := GetMailingJobCategory;
    JobQueueEntry."Maximum No. of Attempts to Run" := MaxNumberOfRetries;
    JobQueueEntry."Record ID to Process" := RecordIdToProcess;
    JobQueueEntry."Parameter String" := CopyStr(ParameterString, 1, MaxStrLen(JobQueueEntry."Parameter String"));
    JobQueueEntry.Description := CopyStr(Description, 1, MaxStrLen(JobQueueEntry.Description));
    //+EVENT
    IsHandled: Boolean;
    OnEnqueueMailingJobOnBeforeEnqueueJobQueueEntry(RecordIdToProcess, ParameterString, Description, JobQueueEntry, IsHandled);
    if not IsHandled then
    //-EVENT
        CODEUNIT.Run(CODEUNIT::"Job Queue - Enqueue", JobQueueEntry);
end;
```

```al
[IntegrationEvent(false, false)]
local procedure OnEnqueueMailingJobOnBeforeEnqueueJobQueueEntry(RecordIdToProcess: RecordID; ParameterString: Text; Description: Text; var JobQueueEntry: Record "Job Queue Entry"; var IsHandled: Boolean)
begin
end;
```

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.
